init.lua

```diff
@@ -1,3 +1,43 @@
+local cmd = vim.cmd
+local env = vim.env
+local fn = vim.fn
+local g = vim.g
+local loop = vim.loop
+local opt = vim.opt
+
+-- The default 'project_name' is taken from the parent directory name of
+-- this configuration repo.
+local config_lua_name = loop.fs_realpath(
+    debug.getinfo(1, 'S').source:match("^@?(.*)$")
+  )
+local config_lua_dir = config_lua_name:match("^(.*)/[^/]*$")
+local config_lua_parent = config_lua_dir:match("^.*/([^/]+)$")
+if config_lua_parent == nil then
+  -- This shouldn't be the case but just in case
+  config_lua_parent = "nvim"
+end
+-- Exporting the environment variable $PROJECT_NAME can force to set 'project_name'
+local project_name = fn.exists(env.PROJECT_NAME) and env.PROJECT_NAME or config_lua_parent
+
+local home_dir = loop.os_homedir()
+local xdg_config_home = fn.exists(env.XDG_CONFIG_HOME) and env.XDG_CONFIG_HOME or home_dir .. '/.config'
```

Comment:
Gotta say this is dark magic within NeoVim that I haven't touched before,
so while it looks good to me, I really can't say without testing